src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -388,7 +388,7 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   <Target Name="CheckWasmSdks">
     <Error Text="Emscripten not found, not compiling to WebAssembly. To enable WebAssembly compilation, install Emscripten and ensure the EMSDK environment variable points to the directory containing upstream/emscripten/emcc.bat"
-           Condition="'$(EMSDK)' == ''" />
+           Condition="'$(EMSDK)' == '' and '$(_targetOS)' == 'browser'" />
     <Error Text="Wasi SDK not found, not compiling to WebAssembly. To enable WebAssembly compilation, install Wasi SDK and ensure the WASI_SDK_PATH environment variable points to the directory containing share/wasi-sysroot"
            Condition="'$(WASI_SDK_PATH)' == '' and '$(_targetOS)' == 'wasi'" />
   </Target>
```

Comment:
It would be good to add a warning for older SDKs that we know will likely not work.

Reply:
Can you elaborate on how we could check this?  Ideally we'd extract the version number from the first line of `${WASI_SDK_PATH}/VERSION` and check it against a "known-to-be-compatible" list, but I'm new to the MSBuild project file DSL, so not sure how do to that here.

Reply:
Is `${WASI_SDK_PATH}/VERSION` a new thing - I don't see it in the v20 layout?

Reply:
Yeah, looks like it's new: https://github.com/WebAssembly/wasi-sdk/commit/0409839729fb4348764905a7e8c2bb53000087a3.  I guess the fallback option would be to run `${WASI_SDK_PATH}/bin/clang --version` and infer the WASI-SDK version from the LLVM version, assuming the latter maps uniquely to the former.

Reply:
Since it looks like the version file is only present in SDK 22, we can just key off of it existing (for now).

I. e. something like:
```
<Warning Condition="!Exists('$(WASI_SDK_PATH)/VERSION')" Text="The WASI SDK version is too low. Please use WASI SDK 22 or newer with a 64 bit Clang." />
```
(I added the 64 bit breadcrumb since it looks like the 22 releases have both 32 bit and 64 bit versions.)